decode/clarity.ts

```diff
@@ -1,40 +1,50 @@
 import version from "../src/core/version";
-import { Event, IAugmentation, IDecodedEvent, IDecodedPayload, IPayload, Token } from "../types/data";
-import data from "./data";
-import diagnostic from "./diagnostic";
-import envelope from "./envelope";
-import interaction from "./interaction";
+import { Augmentation, Event, Payload, Token } from "../types/data";
+import * as Decode from "../types/decode";
+import * as data from "./data";
+import * as diagnostic from "./diagnostic";
+import * as interaction from "./interaction";
 import * as layout from "./layout";
-import metric from "./metric";
 import * as r from "./render";
-import * as summary from "./summary";
 
 let pageId: string = null;
 
-export function decode(input: string | IPayload, augmentations: IAugmentation = null): IDecodedPayload {
-    let json: IPayload = typeof input === "string" ? JSON.parse(input) : input;
-    let timestamp = augmentations ? augmentations.timestamp : Date.now();
-    let ua = augmentations ? augmentations.ua : (navigator && "userAgent" in navigator ? navigator.userAgent : "");
-    let payload: IDecodedPayload = { timestamp, ua, envelope: envelope(json.e), metrics: metric(json.m), analytics: [], playback: [] };
+export function decode(input: string | Payload, augmentations: Augmentation = null): Decode.DecodedPayload {
+    let json: Payload = typeof input === "string" ? JSON.parse(input) : input;
+    let envelope = data.envelope(json.e);
+    let timestamp = augmentations && augmentations.timestamp ? augmentations.timestamp : Date.now();
+    let ua = augmentations && augmentations.ua ? augmentations.ua : (navigator && "userAgent" in navigator ? navigator.userAgent : "");
+    let payload: Decode.DecodedPayload = { timestamp, envelope };
+    if (envelope.sequence === 1) { payload.ua = ua; }
```

Comment:
This is a good change. We don't need to send the UA on each payload. Might consider just making this part of the page event as it isn't really payload property (logically).

Reply:
Moved it to Event.Page instead of augmentations. Also, removed the concept of augmentations for now.